pandas/tests/indexing/test_indexing.py

```diff
@@ -1110,3 +1110,77 @@ def test_setitem_categorical():
         {"h": pd.Categorical(["m", "n"]).reorder_categories(["n", "m"])}
     )
     tm.assert_frame_equal(df, expected)
+
+
+def test_setitem_EA_column_update():
+    # https://github.com/pandas-dev/pandas/issues/33457
+
+    df = pd.DataFrame(
+        {
+            "int": [1, 2, 3],
+            "int2": [3, 4, 5],
+            "float": [0.1, 0.2, 0.3],
+            "EA": pd.array([1, 2, None], dtype="Int64"),
+        }
+    )
+    original_arr = df.EA.array
+
+    # overwrite column with new array
+    df["EA"] = pd.array([1, 2, 3], dtype="Int64")
+    # ensure original array was not modified
+    assert original_arr is not df.EA.array
+    expected = pd.array([1, 2, None], dtype="Int64")
+    tm.assert_extension_array_equal(original_arr, expected)
+
+
+def test_getitem_EA_no_copy():
+    # ensure we don't copy the EA when taking a subset
+
+    df = pd.DataFrame(
+        {
+            "int": [1, 2, 3],
+            "int2": [3, 4, 5],
+            "float": [0.1, 0.2, 0.3],
+            "EA": pd.array([1, 2, None], dtype="Int64"),
+        }
+    )
+    original_arr = df.EA.array
+    subset = df[["int", "EA"]]
+    assert subset.EA.array is original_arr
+    # check that they view the same data by modifying
+    df["EA"].array[0] = 10
+    expected = pd.array([10, 2, None], dtype="Int64")
+    tm.assert_extension_array_equal(subset["EA"].array, expected)
+
+    # TODO this way it doesn't modify subset - is this expected?
```

Comment:
I'm unsure here. The (or an) issue is that we allow setting to change the dtype, which for EAs means changing the *type*. So there's no way for 

```python
s = pd.Series(pd.array([1, 2]))
s.iloc[0] = 0.5
```

to mutate the array inplace, and so I wouldn't expect your `subset` to be mutated.

Reply:
That example currently raises an error, because mutating inplace for EAs doesn't allow changing the dtype (or at least for the nullable EAs, and which is a difference with normal dtypes that we should actually decide if this is desired (I think so) and if so, test and document as a difference)